src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1666,7 +1666,12 @@ void ShenandoahHeap::on_cycle_start(GCCause::Cause cause, ShenandoahGeneration*
   set_gc_cause(cause);
   set_gc_generation(generation);
 
-  generation->heuristics()->record_cycle_start();
+  if (mode()->is_generational()) {
+    // young-gen heuristics track young, bootstrap, and global GC cycle times
```

Comment:
This seems like an unrelated change. Mixing in bootstrap and global gc cycle times is likely to increase the average time and make the heuristics more aggressive.

Reply:
Good catch. I will remove this code from this PR.  (It will come back in a subsequent PR, to support adaptive old evac ratios.  There, we consolidate the accounting for all gc in the "young heuristics", but we parameterize each phase so as to not bias triggering.  Marking phase time is parameterized with bytes marked, for example,